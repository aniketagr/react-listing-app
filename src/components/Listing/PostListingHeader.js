import React from 'react'

function PostListingHeader() {
    return (
        <React.Fragment>
            <h1>Listing Post </h1>
        </React.Fragment>
    )
}

export default PostListingHeader
