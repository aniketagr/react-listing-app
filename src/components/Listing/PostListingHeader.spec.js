import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import PostListingHeader from './PostListingHeader';

beforeEach(() => {
    render(<PostListingHeader />);
});

describe('Post listing header component', () => {
    test('Renders listing post header', () => {
        const linkElement = screen.getByText(/Listing Post/i);
        expect(linkElement).toBeInTheDocument();
    });

    test('Post listing header snapshot', () => {
        const tree = renderer.create(<PostListingHeader />);
        expect(tree.root.findByType("h1").children[0]).toEqual("Listing Posts");
        expect(tree.toJSON()).toMatchSnapshot();
    });
});





















// describe('Post listing header component', () => {
//     it('Renders listing post header', () => {
//         const linkElement = screen.getByText(/Listing Post/i);
//         expect(linkElement).toBeInTheDocument();
//     });

//     it('Post listing header snapshot', () => {
//         const tree = renderer.create(<PostListingHeader />);
//         expect(tree.root.findByType("h1").children[0]).toEqual("Listing Posts");
//         expect(tree.toJSON()).toMatchSnapshot();
//     });
// });
