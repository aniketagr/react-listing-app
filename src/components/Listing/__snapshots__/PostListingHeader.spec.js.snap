// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post listing header component Post listing header snapshot 1`] = `
<h1
  style={
    Object {
      "textAlign": "center",
    }
  }
>
  Listing Posts
</h1>
`;
